pkg/storage/unified/sql/backend.go

```diff
@@ -141,22 +141,30 @@ func (b *backend) WriteEvent(ctx context.Context, event resource.WriteEvent) (in
 func (b *backend) Namespaces(ctx context.Context) ([]string, error) {
 	var namespaces []string
 
-	err := b.db.WithTx(ctx, RepeatableRead, func(ctx context.Context, tx db.Tx) error {
-		rows, err := tx.QueryContext(ctx, "SELECT DISTINCT(namespace) FROM resource ORDER BY namespace;")
-		if err != nil {
-			return err
+	err := b.db.WithTx(ctx, RepeatableRead, func(ctx context.Context, tx db.Tx) (txErr error) {
+		var rows db.Rows
+
+		rows, txErr = tx.QueryContext(ctx, "SELECT DISTINCT(namespace) FROM resource ORDER BY namespace;")
+		if txErr != nil {
+			return txErr
 		}
+
+		defer func() {
+			txErr = rows.Close()
```

Comment:
```suggestion
			_ = rows.Close()
```
This should not overwrite the error. We can just discard it, we don't really care about it.

Reply:
Very good catch! Since we don't care about these errors, I undid some of the changes in the function to simplify it!